### PR TITLE
db(ticdc): add log for debug the flaking test

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5798

### What is changed and how it works?
I can't reproduce it locally and would like to add some logs to see what data is being scanned incorrectly.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by zhaoxinyu — APPROVED** [review]

File: pkg/db/pebble_test.go [hunk 1]
```diff
@@ -29,33 +28,34 @@ func TestIteratorWithTableFilter(t *testing.T) {
...
-		b.Commit()
+		if err := b.Commit(); err != nil {
+			t.Errorf("Put failed: %v", err)
+		}
 	}
 	if err = db.(*pebbleDB).db.Flush(); err != nil {
-		errmsg := fmt.Sprintf("Flush fail: %v", err)
-		panic(errmsg)
+		t.Errorf("Flush failed: %v", err)
 	}
```
Comment (hicqu) [inline pkg/db/pebble_test.go:60]: I suggest to add this:
```go
    // your code...

    // There should be no any compactions but 2 tables at L0.
    pebbleInstance := db.(*pebbleDB).db
    stats := pebbleInstance.Metrics()
    require.Equal(t, int64(0), stats.Compact.Count)
    require.Equal(t, int64(2), stats.Levels[0].NumFiles)
    for level := 1; level < 7; level++ { // 7 is a pebble internal constant.
        require.Equal(t, int64(0), stats.Levels[level].NumFiles)
    }
    
    // your code...
```

**Comment by hicqu** [comment]
LGTM

**Review by nongfushanquan — APPROVED** [review]

**Comment by hi-rustin** [comment]
test failed:
```
[2022-06-27T13:01:19.673Z] === FAIL: pkg/db TestIteratorWithTableFilter (0.04s)

[2022-06-27T13:01:19.673Z]     pebble_test.go:84: lowerTs: 0, upperTs: 1, expectedCount: 7

[2022-06-27T13:01:19.673Z]     pebble_test.go:98: uniqueID: 1, tableID: 1, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.673Z]     pebble_test.go:98: uniqueID: 1, tableID: 2, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.673Z]     pebble_test.go:98: uniqueID: 1, tableID: 3, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.673Z]     pebble_test.go:98: uniqueID: 1, tableID: 4, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.673Z]     pebble_test.go:98: uniqueID: 1, tableID: 5, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 6, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 7, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:84: lowerTs: 1, upperTs: 2, expectedCount: 7

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 1, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 2, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 3, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 4, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 5, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 6, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 7, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:84: lowerTs: 2, upperTs: 3, expectedCount: 9

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 1, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 1, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 2, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 2, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 3, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 3, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 4, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 4, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 5, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 5, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 6, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 6, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 7, startTs: 0, CRTs: 1

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 7, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 8, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:98: uniqueID: 1, tableID: 9, startTs: 0, CRTs: 3

[2022-06-27T13:01:19.674Z]     pebble_test.go:102: 

[2022-06-27T13:01:19.674Z]         	Error Trace:	pebble_test.go:102

[2022-06-27T13:01:19.674Z]         	Error:      	Not equal: 

[2022-06-27T13:01:19.674Z]         	            	expected: 9

[2022-06-27T13:01:19.674Z]         	            	actual  : 16

[2022-06-27T13:01:19.674Z]         	Test:       	TestIteratorWithTableFilter

```
https://ci.pingcap.net/blue/organizations/jenkins/ut-check/detail/ut-check/19563/pipeline/

/hold

**Comment by hi-rustin** [comment]
https://github.com/pingcap/tiflow/pull/6116#pullrequestreview-1023003572

**Pull request closed without merging**